docs/types/standard-linter-v2.md

```diff
@@ -0,0 +1,46 @@
+# Standard Linter v2
+
+This document describes the type of Standard Linter v2, for an example in how to implement this in your own package see the see the [Example Usage](../examples/standard-linter-v2.md) document.
+
+## Type
+
+```js
+type Linter = {
+  // Automatically added, not guaranteed
+  __$sb_linter_version: 2,
+  __$sb_linter_activated: boolean,
+  __$sb_linter_request_latest: number,
+  __$sb_linter_request_last_received: number,
+
+  // From providers
+  name: string,
+  scope: 'file' | 'project',
+  lintOnFly: boolean,
+  grammarScopes: Array<string>,
+  lint(textEditor: TextEditor): ?Array<Message | MessageLegacy> | Promise<?Array<Message | MessageLegacy>>,
+}
+```
+
+## FAQs
+
+**Q**: What's the difference between `file` and `project` scopes?
+
+**A**: Their newer results only replace their previous results for that particular file. Their newer results only replace their previous results for that particular file. Examples of such linters are `ESLint`.
+
+Results of `project` scoped linters are not tied to any TextEditor and are not discarded even when all TextEditors are disposed. Their new results replace all of their previous results. This is useful for linters that run background daemons that scan the entire project, `FlowType` for example.
```

Comment:
"Their new results replace all of their previous results.", again for a standard linter provider this is just meaningless internals to how Linter works and should just be removed.